qiskit/dagcircuit/dagcircuit.py

```diff
@@ -73,7 +73,7 @@ class DAGCircuit:
 
     # pylint: disable=invalid-name
 
-    def __init__(self):
+    def __init__(self, _node_count_hint=0, _edge_count_hint=0):
```

Comment:
Is there any harm in making this public?  I don't really like having "private" arguments unless we really need them, and there's no real safety concerns with these.

Reply:
Not really, I was mostly worried about making it public because I pushed this up after the feature freeze and that would make it a feature. But I think this makes a straightforward exception to the process since it's so simple.

Reply:
Yeah, I'm fine with allowing it, and I'd rather just make the things public than have awkward "private" arguments show up in the documentation.

Reply:
Made public in: https://github.com/Qiskit/qiskit/pull/12812/commits/eb91a48c2e3478577cf0f3494f23ee1397fe9384